packages/tempus/src/Tempus_RKButcherTableau.hpp

```diff
@@ -1678,89 +1793,28 @@ class GeneralDIRK_RKBT :
   getValidParameters() const
   {
     Teuchos::RCP<Teuchos::ParameterList> pl = Teuchos::parameterList();
-    pl->setParameters(*(RKButcherTableau<Scalar>::getValidParameters()));
+    pl->setParameters(*(RKButcherTableau<Scalar>::getValidParametersImplicit()));
     pl->set<bool>("Initial Condition Consistency Check", false);
 
     // Tableau ParameterList
+    using Teuchos::as;
     typedef Teuchos::ScalarTraits<Scalar> ST;
-    const Scalar one = ST::one();
-    std::string gamma = std::to_string(Teuchos::as<Scalar>((2*one-ST::squareroot(2*one))/(2*one)));
-    std::string one_gamma = std::to_string(Teuchos::as<Scalar>(one-(2*one-ST::squareroot(2*one))/(2*one)));
-    Teuchos::RCP<Teuchos::ParameterList> tableauPL = Teuchos::parameterList();
-    tableauPL->set<std::string>("A", gamma + " 0.0; " + one_gamma + " "+gamma);
-    tableauPL->set<std::string>("b", one_gamma + " " + gamma);
-    tableauPL->set<std::string>("c", gamma + " 1.0");
-    tableauPL->set<int>("order", 2);
-    pl->set("Tableau", *tableauPL);
-
-    return pl;
-  }
-};
-
-
-// ----------------------------------------------------------------------------
-/** \brief SDIRK 1 Stage 1st order
- *
- *  The tableau (order=1) (also known as Backward Euler) is
- *  \f[
- *  \begin{array}{c|c}
- *    c & A \\ \hline
- *      & b^T
- *  \end{array}
- *  \;\;\;\;\mbox{ where }\;\;\;\;
- *  \begin{array}{c|c} 1 & 1 \\ \hline
- *                       & 1 \end{array}
- *  \f]
- */
-template<class Scalar>
-class SDIRK1Stage1stOrder_RKBT :
-  virtual public RKButcherTableau<Scalar>
-{
-  public:
-  SDIRK1Stage1stOrder_RKBT()
-  {
-    std::stringstream Description;
-    Description << this->description() << "\n"
-                << "c = [ 1 ]'\n"
-                << "A = [ 1 ]\n"
-                << "b = [ 1 ]'";
-
-    this->setDescription(Description.str());
-    this->setParameterList(Teuchos::null);
-  }
-
-  virtual std::string description() const { return "SDIRK 1 Stage 1st order"; }
-
-  void setParameterList(Teuchos::RCP<Teuchos::ParameterList> const& pList)
-  {
-    RKButcherTableau<Scalar>::setParameterList(pList);
-
-    typedef Teuchos::ScalarTraits<Scalar> ST;
-    int NumStages = 1;
+    int NumStages = 2;
     Teuchos::SerialDenseMatrix<int,Scalar> A(NumStages,NumStages);
     Teuchos::SerialDenseVector<int,Scalar> b(NumStages);
     Teuchos::SerialDenseVector<int,Scalar> c(NumStages);
 
-    // Fill A:
-    A(0,0) = ST::one();
-
-    // Fill b:
-    b(0) = ST::one();
-
-    // Fill c:
-    c(0) = ST::one();
-
-    int order = 1;
-
-    this->initialize(A,b,c,order,this->getDescription());
-  }
+    const Scalar one   = ST::one();
+    const Scalar gamma = as<Scalar>((2*one-ST::squareroot(2*one))/(2*one));
```

Comment:
Be careful with `operator* (int, Scalar)`.  It does not exist for complex Scalar types, for example.  See the following example: https://godbolt.org/z/kea6Im